src/Engines/ElasticsearchEngine.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+namespace Laravel\Scout\Engines;
+
+use Laravel\Scout\Builder;
+use Elasticsearch\Client as Elasticsearch;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Support\Collection as BaseCollection;
+
+class ElasticsearchEngine extends Engine
+{
+    /**
+     * @var string $index
+     */
+    protected $index;
+
+    /**
+     * Create a new engine instance.
+     *
+     * @param  \Elasticsearch\Client  $elasticsearch
+     * @return void
+     */
+    public function __construct(Elasticsearch $elasticsearch, $index)
+    {
+        $this->elasticsearch = $elasticsearch;
+
+        $this->index = $index;
+    }
+
+    /**
+     * Update the given model in the index.
+     *
+     * @param  Collection  $models
+     * @return void
+     */
+    public function update($models)
+    {
+        $models->each(function ($model) {
+            $this->elasticsearch->index([
+                'index' => $this->index,
+                'type' => $model->searchableAs(),
+                'id' => $model->getKey(),
+                'body' => $model->toSearchableArray(),
+            ]);
+        });
+    }
+
+    /**
+     * Remove the given model from the index.
+     *
+     * @param  Collection  $models
+     * @return void
+     */
+    public function delete($models)
+    {
+        $models->each(function ($model) {
+            $this->elasticsearch->delete([
```

Comment:
@hernandev i think he means that instead of each the bulk operation should be used.
https://www.elastic.co/guide/en/elasticsearch/guide/current/bulk.html